### PR TITLE
3 get user

### LINKED ISSUE #3: Get user
`GET /api/v1/users/:id`

- [ ] Returns object of user data
- [ ] Returns all of that users ideas
- [ ] Returns all of that users comment
- [ ] Returns all of that users contributions
- [ ] Returns a 404 if the user cannot be found

```javascript
{
  id: 123,
  uid: aa123456u,
  email: 'notanemail@na.moc',
  username: 'jude-lawson',
  ideas: [...],
  comments: [...],
  contributions: [...]
}
```

### PR DESCRIPTION
**Pull request by jude-lawson**
@jude-lawson Wrote Tests
@jude-lawson Implemented 
@zdcthomas  Reviewed

# Neccesary checkmarks:
- [X] All Tests are Passing
- [ ] The code will run locally

## Type of change
- [X] New feature 
- [ ] Bug Fix

# Implements/Fixes:
- Implements Users#show action
  - Returns a JSON object with the following attributes:
    - id
    - uid
    - email
    - username
    - ideas
    - contributions
    - comments
- Returns a 404 with error messages if the user cannot be found
closes #3 

# Check the correct boxes
- [X] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

# Testing Changes
- [X] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:
- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

# Notes:

# Code Snippets of Changes:
```

```

(╯°□°）╯︵ ┻━┻

### CODE REVIEW

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :uid, :email, :username, :ideas, :contributions, :comments
+
+  def ideas
```
Comment (zdcthomas) [inline app/serializers/user_serializer.rb]: you could try the has_many :ideas form for less verbosity, but it isn't a big deal

File: config/routes.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
+      # Ideas
```
Comment (zdcthomas) [inline config/routes.rb]: 👍

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -28,11 +28,27 @@
   config.filter_rails_from_backtrace!
 
   config.before :each do
-    DatabaseCleaner.start
```
Comment (zdcthomas) [inline spec/rails_helper.rb]: Was database cleaner not working as expected?/What's the benefit of the manual db clearance?

Comment (jude-lawson) [reply spec/rails_helper.rb]: Should be all good now 👍

**Comment by zdcthomas** [comment]
This is good, i think it will be less verbose and more open to future changes. Thanks

**Pull request merged**